pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -295,19 +291,20 @@ func (exp *flowExporter) sendFlowRecords() error {
 				}
 				exp.numDataSetsSent = exp.numDataSetsSent + 1
 			}
-			// If the connection is not present in the conntrack table anymore,
-			// then we delete the connection in connection store and record store.
+			// If connection is in dying state, we will delete the connection in
+			// connection store and record store.
 			// Please note that our record cache policy is that we keep the record in
 			// the record map as long as the corresponding connection is present in
 			// the conntrack table.
-			if !record.Conn.IsPresent {
+			if flowexporter.IsConnectionDying(&record.Conn) {
 				klog.V(2).Infof("Deleting the inactive connection with key: %v from both connection map and record map", key)
 				if err := exp.connStore.DeleteConnectionByKey(key); err != nil {
 					return err
 				}
 				if err := exp.flowRecords.DeleteFlowRecordWithoutLock(key); err != nil {
 					return err
 				}
+				record.Conn.IsExportedAndRemoved = true
```

Comment:
We delete both connection and record (with connection copy) for the corresponding key in the above two lines. Will setting this help? 
Were you able to validate this through a unit test?

Reply:
That's true. It does not help. I have it removed. Do you have any suggestion to handle the connection comes with TIME_WAIT state then? Is it good for us not sending the record?

Reply:
As we are polling at an interval and we may miss states, it is definitely a valid concern.
How about delaying the deletion of connection from the connection map until it is seen next time and the record is already sent (we need a field in the connection of connection map and it has to be set separately)? This means we should move the deletion logic of connection from exporter.go to connections.go, which I feel is the right place. 

Above scenario for flows which come in with "TIME_WAIT" state is covered.

Reply:
In this case we still cannot avoid re-adding the connection that has been exported and deleted, right?

Reply:
After offline discussion, the optimization is planned as follows:
1) handle connection deletion from connection map in `resetConn` from `Poll()` when the connection no longer exists in conntrack table (`connections.go`)
2) delete flow records of corresponding dying connection after exporting, and set `isExportedAndDying = true` to corresponding connection (`exporter.go`)
3) avoid adding/updating both connection map and flow records map when `isExportedAndDying` of the connection  is set to true